### DISCUSSION TITLE
v2.51，点"发送链接到Aria2"未推送成功

### DISCUSSION THREAD

**Discussion by jutss** [AlistGo/alist#1088] [Q&A]
你好，目前碰到1个小问题：
选中文件，复制直链后，打开vps上装好的AriaNg（ariang.js.org 的前端），新建任务后都可以正常下载。
但是选择**发送链接到Aria2**，网页提示“已发送”，但实际Aria2未接收到。
在AriaNg里面看不到这个记录，文件也未下载到vps硬盘。

Alist版本：v2.5.1
Aria2版本：1.36
VPS系统：centos7

在Alist后端已经配置好Aria2 RPC url  （格式是http://vps的IP地址:6800/jsonrpc）
以及Aria2 RPC secret都填入了。

默认设置是下载完成后存在vps硬盘里。

Aria2安装是通过一键脚本：https://github.com/P3TERX/aria2.sh
安装完后配置目录是在/.aria2**c**/下，不是/.aria2，不知道是不是这个原因...

**Comment by miniers** [comment]
如果alist是https，那么也需要用nginx来转发aria2的rpc为https的

**Reply by jutss** [reply]
你好，谢谢回复！域名的确是申请了ssl证书，但是nginx我是小白，下面是现在的配置，不知道该咋改.../(ㄒoㄒ)/

```
server {
    listen 80;
    server_name 绑定的域名 www.绑定的域名;
    return 301 https://$server_name$request_uri;   
}


server {
        listen 443 ssl;
        server_name 绑定的域名 www.绑定的域名;
        ssl_certificate "密钥目录";
        ssl_certificate_key "密钥目录";

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Range $http_range;
        proxy_set_header If-Range $http_if_range;
        proxy_redirect off;
        proxy_pass http://127.0.0.1:5244;
        # 上传的最大文件尺寸
        client_max_body_size 20000m;
    }
}
```

**Reply by creeper2020** [reply]
server
    {
        listen 80;
        #listen [::]:80;
        server_name 域名 ;
        index index.html index.htm index.php default.html default.htm default.php;
        root  /home/wwwroot/;

        include rewrite/none.conf;
        #error_page   404   /404.html;

        # Deny access to PHP files in specific directory
        #location ~ /(wp-content|uploads|wp-includes|images)/.*\.php$ { deny all; }

        include enable-php.conf;
 location / {
    proxy_pass  http://127.0.0.1:6800; # 转发规则
    proxy_set_header Host $proxy_host; # 
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
       


server
    {
        listen 443 ssl http2;
        #listen [::]:443 ssl http2;
        server_name 域名 ;
        index index.html index.htm index.php default.html default.htm default.php;
        root  /home/wwwroot/;

        ssl_certificate /usr/local/nginx/conf/ssl//fullchain.cer;#修改为你的证书
        ssl_certificate_key /usr/local/nginx/conf/ssl/.key;
        ssl_session_timeout 5m;
        ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
        ssl_prefer_server_ciphers on;
        ssl_ciphers "TLS13-AES-256-GCM-SHA384:TLS13-CHACHA20-POLY1305-SHA256:TLS13-AES-128-GCM-SHA256:TLS13-AES-128-CCM-8-SHA256:TLS13-AES-128-CCM-SHA256:EECDH+CHACHA20:EECDH+CHACHA20-draft:EECDH+AES128:RSA+AES128:EECDH+AES256:RSA+AES256:EECDH+3DES:RSA+3DES:!MD5";
        ssl_session_cache builtin:1000 shared:SSL:10m;
        # openssl dhparam -out /usr/local/nginx/conf/ssl/dhparam.pem 2048
        ssl_dhparam /usr/local/nginx/conf/ssl/dhparam.pem;

        include rewrite/none.conf;
        #error_page   404   /404.html;

        # Deny access to PHP files in specific directory
        #location ~ /(wp-content|uploads|wp-includes|images)/.*\.php$ { deny all; }

        include enable-php.conf;

        
 location / {
    proxy_pass  http://127.0.0.1:6800; # 转发规则
    proxy_set_header Host $proxy_host; # 
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
直接抄我的配置就行，如果你不用80端口的话把80删了也行
